src/common/create-ignorer.js

```diff
@@ -2,40 +2,80 @@
 
 const path = require("path");
 const ignore = require("ignore").default;
+
 const getFileContentOrNull = require("../utils/get-file-content-or-null.js");
+const thirdParty = require("../common/third-party.js");
 
 /**
+ * @param {string} filePath
  * @param {string?} ignorePath
  * @param {boolean?} withNodeModules
  */
-async function createIgnorer(ignorePath, withNodeModules) {
-  const ignoreContent = ignorePath
-    ? await getFileContentOrNull(path.resolve(ignorePath))
-    : null;
+async function createIgnorer(filePath, ignorePath, withNodeModules) {
+  // prettier-ignore
+  const ignoreContent = ignorePath &&
+    await getFileContentOrNull(path.resolve(ignorePath)) ||
+    await _createExplorer(false).search(path.resolve(filePath));
 
   return _createIgnorer(ignoreContent, withNodeModules);
 }
 
 /**
+ * @param {string} filePath
  * @param {string?} ignorePath
  * @param {boolean?} withNodeModules
  */
-createIgnorer.sync = function (ignorePath, withNodeModules) {
-  const ignoreContent = !ignorePath
-    ? null
-    : getFileContentOrNull.sync(path.resolve(ignorePath));
+createIgnorer.sync = function (filePath, ignorePath, withNodeModules) {
```

Comment:
Let's change base to `next` branch, so we don't need support sync version.

Reply:
Done.